client/highlevel-rest/src/test/java/org/elasticsearch/client/HLClientSearchIT.java

```diff
@@ -39,34 +53,90 @@ public void init() {
     }
 
     public void createTestDoc() throws IOException {
+        XContentBuilder mappingsBuilder = randomXContentBuilder();
+        mappingsBuilder.startObject();
+        mappingsBuilder.startObject("mappings");
+        mappingsBuilder.startObject("type");
+        mappingsBuilder.startObject("properties");
+        mappingsBuilder.startObject("title");
+        mappingsBuilder.field("type", "text");
+        mappingsBuilder.field("store", "true");
+        mappingsBuilder.endObject();
+        mappingsBuilder.startObject("content");
+        mappingsBuilder.field("type", "text");
+        mappingsBuilder.field("store", "true");
+        mappingsBuilder.endObject();
+        mappingsBuilder.endObject();
+        mappingsBuilder.endObject();
+        mappingsBuilder.endObject();
+        mappingsBuilder.endObject();
```

Comment:
if we'll need this in other tests, we should probably try to shorten this setup part of test by re-using what we have in our java api, that allows to provide `Object... source` , but we also want to be able to randomize the xcontent type, which is why we need to adapt it a bit